plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageService.java

```diff
@@ -216,8 +234,13 @@ protected PasswordAuthentication getPasswordAuthentication() {
      * <a href="https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/migrationGuides/V8_V12.md#miscellaneous">migration guide</a> for mode details:
      */
     private BlobServiceClientBuilder applyLocationMode(final BlobServiceClientBuilder builder, final AzureStorageSettings settings) {
-        final StorageConnectionString storageConnectionString = StorageConnectionString.create(settings.getConnectString(), logger);
-        final StorageEndpoint endpoint = storageConnectionString.getBlobEndpoint();
+        final StorageEndpoint endpoint;
+        if (settings.usesManagedIdentityCredential()) {
```

Comment:
Please add method getEndpoint() to the `AzureStorageSettings`, which will be returned based on the identity configuration:

```
final StorageEndpoint endpoint = settings.getEndpoint();
```

Reply:
@reta After some consideration. I have decided to keep the class variable `connectString`, and implemented a `getEndpoint()` to the AzureStorageSettings. This would ensures forward compatibility since when getting storage endpoint using connection string was previously executed at run time.

This is the [commit change](https://github.com/opensearch-project/OpenSearch/pull/12559/commits/ae7dd99c2e998062b897192ce82367245f3f60dd).

Reply:
> @reta After some consideration. I have decided to keep the class variable `connectString`, and implemented a `getEndpoint()` to the AzureStorageSettings.

@chengwushi-netapp  I don't understand the issue: as per implementation,  `connectionString` has no use with managed identity, this property should not be exposed / used directly anymore.